dev/fattest.simplicity/src/com/ibm/websphere/simplicity/config/MpOpenAPIElement.java

```diff
@@ -46,9 +66,153 @@ public void setUiPath(String uiPath) {
+    }
+
+    /**
+     * @return a list of modules to be excluded
+     */
+    public List<String> getExcludedModules() {
+        return (excludedModules == null) ? (excludedModules = new ArrayList<String>()) : excludedModules;
+    }
+
     @Override
     public String toString() {
-        return "MpOpenAPIElement [docPath=" + docPath + ", uiPath=" + uiPath + "]";
+
+        StringBuilder sb = new StringBuilder();
+        sb.append(", includeApplication=").append("[" + String.join(",", getIncludedApplications()) + "]");
+        sb.append(", excludeApplication=").append("[" + String.join(",", getExcludedApplications()) + "]");
+        sb.append(", includeModule=").append("[" + String.join(",", getIncludedModules()) + "]");
+        sb.append(", excludeModule=").append("[" + String.join(",", getExcludedModules()) + "]");
+
+        return "MpOpenAPIElement [docPath=" + docPath + ", uiPath=" + uiPath + sb.toString() + "]";
+    }
+
+    public static class MpOpenAPIElementBuilder {
+
+        private final static Logger LOG = Logger.getLogger("MpOpenAPIElementBuilder");
+
+        public static MpOpenAPIElementBuilder cloneBuilderFromServer(LibertyServer server) throws CloneNotSupportedException, Exception {
+            return new MpOpenAPIElementBuilder(server);
+        }
+
+        public static MpOpenAPIElementBuilder cloneBuilderFromServerResetAppsAndModules(LibertyServer server) throws CloneNotSupportedException, Exception {
+            MpOpenAPIElementBuilder builder = new MpOpenAPIElementBuilder(server);
+            nullSafeClear(builder.element.excludedApplications);
+            nullSafeClear(builder.element.includedApplications);
+            nullSafeClear(builder.element.excludedModules);
+            nullSafeClear(builder.element.excludedModules);
+            return builder;
+        }
+
+        private final MpOpenAPIElement element;
+        private final LibertyServer server;
+        private final ServerConfiguration serverConfig;
+
+        private MpOpenAPIElementBuilder(LibertyServer server) throws CloneNotSupportedException, Exception {
+            this.server = server;
+            this.serverConfig = server.getServerConfiguration().clone();
+            this.element = serverConfig.getMpOpenAPIElement();
+        }
+
+        public void buildAndPushToServer() throws Exception {
+            LOG.info("Pushing new server configuration: " + serverConfig.toString());
+            server.updateServerConfiguration(serverConfig);
+            if (server.isStarted()) {
+                server.setMarkToEndOfLog();
+                server.waitForStringInLogUsingMark("CWWKG0017I"); //The server configuration was successfully updated
+                //Setting a low timeout because this only appears if OpenAPI trace is enabled (and enabling trace breaks debuggers)
+                server.waitForStringInLogUsingMark("Finished creating OpenAPI provider", 500);
+            }
+        }
```

Comment:
I don't really like having this here. In general the classes within this package are concerned with modelling the configuration and not with reading or writing the config to the server.

Let's leave it for now and revise it in a future PR.